### PR TITLE
feat(config): template more aeotec devices

### LINKED ISSUE #2634: Add Aeotec Smart Switch 7
This seems premature until the config from @blhoward2 is available publicly as it seems a few people are tracking using this PR.

_Originally posted by @Nixon506E in https://github.com/zwave-js/node-zwave-js/issues/2027#issuecomment-842668658_

### LINKED ISSUE #2637: Add Aeotec ZWA019
**Which device is missing?**

Aeotec Water Sensor 7 Pro
Exists in openzwave database already 0371:0013:0102

**What are the IDs?**

Manufacturer ID: 0x0371
Product type: 0x0102  
Product ID: 0x0013  

**Did you check if we really don't have it?**

- [ ] It is not in the [configuration DB](https://devices.zwave-js.io/)
- [ ] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)
- [ ] It is not in the [TODO list](https://github.com/zwave-js/node-zwave-js/issues/1600)

**Do you have a link to the PDF manual?**

https://products.z-wavealliance.org/ProductManual/File?folder=&filename=product_documents/3951/ZWA019.pdf

### LINKED ISSUE #2664: 🚨🚨🚨 PSA: Please do not provide PRs for Aeotec files 🚨🚨🚨
We appreciate all the help we can get with configuration files, but please **DO NOT** open PRs for new Aeotec configurations. They are currently being worked on by @blhoward2.

### LINKED ISSUE #2796: Missing device configuration params for DSC08 AEON Labs / Smart Energy Illuminator.
**Checklist:**

-   [X] My problem is **not** within Home Assistant or the ZWave JS integration. **Or:** a Home Assistant developer has told me to come here.
-   [X] My problem is **not** within ZWaveJS2MQTT. **Or:** a ZWaveJS2MQTT developer has told me to come here.
-   [X] I have checked the troubleshooting section and my problem is **not** described there.
-   [X] I have read the changelog and my problem was **not** mentioned there.

**Describe the bug**

As [instructed](https://github.com/zwave-js/node-zwave-js/discussions/2781), providing details for the [DSC08 AEON Labs / Smart Energy Illuminator](https://devices.zwave-js.io/?jumpTo=0x0086:0x0003:0x0008:0.0) which provides config params **80, 90, 91 and 92** but these are missing from the associated configuration file and can therefore not be set via Z-Wave JS. 
**Device information**

```
  "manufacturer": "AEON Labs",
  "manufacturerId": 134,
  "productType": 3,
  "productId": 8,
  "firmwareVersion": "1.27",
```

**Did you change anything?**

-   [ ] **Yes**: _(please describe)_
-   [X] **No**

**Did this use to work before?**

-   [X] **Don't know**, this is a new device
-   [ ] **No**, it never worked anywhere
-   [ ] **Yes**, in: _(specify application with versions)_

**How are you using `node-zwave-js`**

-   [ ] `zwavejs2mqtt` (latest) docker image
-   [ ] `zwavejs2mqtt` (dev) docker image
-   [ ] `zwavejs2mqtt` Manual Docker build
    -   `node-zwave-js` branch: 
    -   `zwavejs2mqtt` branch: 
-   [ ] ioBroker.zwave2 adapter
-   [X] `HomeAssistant` version core-2021.6.2
-   [ ] Pkg
-   [ ] Manual Docker build
    -   `node-zwave-js` branch: 
    -   `zwavejs2mqtt` branch: 
-   [ ] `node-red-contrib-zwave-js`
    -   Module version (double click node): 
-   [ ] Manually built (as described in the [docs](https://zwave-js.github.io/node-zwave-js/#/development/installing-from-github))
-   [ ] Other: 

**To Reproduce**

Attempt to set the above missing params in Home Assistant. Error message:
`websocket_api script: Error executing script. Unexpected error for call_service at pos 1: Configuration parameter with value ID 4-112-0-80 could not be found`

-->

### LINKED ISSUE #2893: Aeotec AW111 Dual Nano Dimmer configuration options
a Home Assistant developer has told me to come here.
https://github.com/home-assistant/core/issues/52011

**Describe the bug**
I have several Aeotec AW111 Dual Nano Dimmers. In the legacy Z-Wave integration I was able to set parameters 123 & 124 to control whether the load was triggered by the physical switch inputs. This functionality appears read-only in Zwave-JS.

It looks like the node-zwave-js/packages/config/config/devices/0x0086/templates/aeotec_template.json file has these parameters set to 'read'. (line 1359 & 1383)

(also there seems to be a typo in the S2 destination field - line 1395 should read "label": "Load and association group 4" rather than 3)

**Device information**
Which device(s) is/are affected (manufacturer/model)?  Aeotec AW111
What is/are the node IDs? 32

**Did you change anything?**
Yes - moved to Zwave-JS from legacy Zwave integration

**Did this use to work before?**
Yes, in: legacy Z-wave integration

**How are you using `node-zwave-js`**
HomeAssistant` core-2021.6.5

**To Reproduce**

1. Go to the device page
2. Click on 'Configure Device'
3. Scroll down to '.Control Destonation (S1)'
4. See error 'This parameter is read-only'

**Additional context**
N/A

**Logfile:**
shout if you need the logfile, not sure it's going to help much here!

### PR DESCRIPTION
**Pull request by blhoward2**
Closes #2893
Closes #2796
Closes #2637
Closes #2634
Closed #2664 

After this, I have just three more Aeotec files to go. Bulb 6 Multi-White, Bulb 6 Multi-Color, and Multisensor 7.

### CODE REVIEW

File: packages/config/config/devices/0x0086/templates/aeotec_template.json [hunk 1]
```diff
@@ -606,6 +785,88 @@
...
+		"options": [
+			{
+				"label": "Disable",
+				"value": 0
+			}
+		]
+	},
+	"auto_report_interval_kwh": {
+		"$import": "#base_0-65535",
+		"label": "Automatic Reporting Interval: KwH",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0086/templates/aeotec_template.json]:
```suggestion
		"label": "Automatic Reporting Interval: kWh",
```

File: packages/config/config/devices/0x0086/templates/aeotec_template.json [hunk 2]
```diff
@@ -1069,6 +1330,125 @@
...
+				"label": "Switch after ‘open‘ and ‘close‘",
+				"value": 0
+			},
+			{
+				"label": "Switch after ‘open‘",
+				"value": 1
+			},
+			{
+				"label": "Switch after ‘close‘",
+				"value": 2
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0086/templates/aeotec_template.json]: These quotes look weird.

File: packages/config/config/devices/0x0371/zw187.json [hunk 3]
```diff
@@ -6,96 +6,66 @@
...
-				},
-				{
-					"label": "Closed when magnet is away, open when magnet is near.",
-					"value": 1
-				}
-			]
+			"$import": "../0x0086/templates/aeotec_template.json#invert_state_report"
 		},
 		"3": {
 			"label": "Association Group 2 Settings",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zw187.json]: I think repeating "Basic Set" in each option is a bit redundant. How about this?
```suggestion
			"label": "Association Group 2: Basic Set Value",
```
and then shorten the options.

File: packages/config/config/devices/0x0371/zw187.json [hunk 4]
```diff
@@ -107,84 +77,77 @@
...
-			"unsigned": true,
+			"label": "Retry Wait Time",
 			"valueSize": 2,
+			"unit": "100 ms",
 			"minValue": 0,
 			"maxValue": 255,
-			"defaultValue": 0
+			"defaultValue": 0,
+			"unsigned": true
 		},
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zw187.json]: Should we mention that this applies to the commands the device sends?

File: packages/config/config/devices/0x0371/zw175.json [hunk 5]
```diff
@@ -6,295 +6,234 @@
...
-					"label": "Night Light Mode",
+					"label": "Idle state from corresponding alarm",
 					"value": 1
-				},
-				{
-					"label": "On/Off Mode",
-					"value": 2
 				}
 			]
 		},
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zw175.json]: This parameter confuses me. What does it do? Alarm duration or something else?

File: packages/config/config/devices/0x0371/zw175.json [hunk 6]
```diff
@@ -6,295 +6,234 @@
...
+		"18": {
+			"label": "LED Blinking Frequency",
+			"description": "Defines the number of blinks per second during alarm activation",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zw175.json]: What about this?
```suggestion
			"unit": "Hz",
```

File: packages/config/config/devices/0x0371/zw175.json [hunk 7]
```diff
@@ -6,295 +6,234 @@
...
-			"label": "Night Light disable (minute)",
-			"description": "Enable or disable Night Light Mode during specific times",
+			"label": "Night Light (Disable): Minute",
+			"description": "Allowable range: 0-59",
 			"valueSize": 4,
 			"minValue": 0,
-			"maxValue": 60,
+			"maxValue": 59,
 			"defaultValue": 0
 		},
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zw175.json:184]: Just to make sure - this isn't the weird time as hex thing, right?

File: packages/config/config/devices/0x0371/zwa005.json [hunk 8]
```diff
@@ -108,24 +106,24 @@
 			]
 		},
 		"6": {
-			"label": "Basic Set Value Settings for Group 2",
-			"description": "Values to send to Group 2",
+			"label": "Basic Set Value For Group 2",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa005.json:109]: I commented on another parameter that seems to be the same, but is formulated very differently. Probably makes sense to template that one?

File: packages/config/config/devices/0x0371/zwa005.json [hunk 9]
```diff
@@ -137,304 +135,106 @@
...
+					"label": "Set 0xff on motion cleared",
 					"value": 5
 				}
 			]
 		},
 		"7": {
-			"label": "Temperature Alarm Value",
-			"description": "Alert nodes when temperature is above this value",
-			"unit": "0.1 degrees (C/F)",
+			"label": "Temperature Alarm Threshhold (High)",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa005.json]:
```suggestion
			"label": "Temperature Alarm Threshold (High)",
```

File: packages/config/config/devices/0x0371/zwa005.json [hunk 10]
```diff
@@ -137,304 +135,106 @@
...
-			"defaultValue": 3600
+			"defaultValue": 3600,
+			"unsigned": true
 		},
-		"24": {
-			"label": "Light Sensor Report Interval",
-			"description": "Interval for timed light level reports",
-			"unit": "Seconds",
+		"18": {
+			"label": "Timed Light Sensor Report Interval",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa005.json]: I think we should define somewhere how to call the automatic/timed/whatever reports, so we can make it consistent everywhere.

File: packages/config/config/devices/0x0371/zwa009.json [hunk 11]
```diff
@@ -17,39 +17,37 @@
...
-			"label": "Low Humidity Notification",
+			"label": "Low Humidity",
 			"maxNodes": 5
 		},
 		"6": {
-			"label": "Air Temperature",
-			"description": "A multilevel sensor report is sent to the nodes in this group",
+			"label": "Air Temperature (MultiLevel)",
 			"maxNodes": 5
 		}
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa009.json:45]: If I understand correctly, 2-5 send Notifications, 6 sends a Multilevel Sensor Report. Depending on what you want to do with the associations, having that info at a glance would be helpful I think

File: packages/config/config/devices/0x0371/zwa009.json [hunk 12]
```diff
@@ -17,39 +17,37 @@
...
 	},
 	"paramInformation": {
 		"1": {
-			"label": "Minimum Temperature Change to Report",
-			"unit": "1/10 degree",
+			"label": "Temperature Change Report Threshhold",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa009.json]:
```suggestion
			"label": "Temperature Change Report Threshold",
```
or am I mistaken here? Same happens in a few more location.

File: packages/config/config/devices/0x0371/zwa021.json [hunk 13]
```diff
@@ -53,106 +60,113 @@
...
+					"label": "Disable",
 					"value": 0
 				}
 			]
 		},
 		"6": {
-			"label": "Valve position",
-			"description": "Valve position report. Reporting delta in percent. '0' to disable reporting.",
-			"unit": "%",
+			"label": "Valve position Report",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa021.json]:
```suggestion
			"label": "Valve Position Report",
```

File: packages/config/config/devices/0x0371/zwa021.json [hunk 14]
```diff
@@ -53,106 +60,113 @@
...
 				},
 				{
-					"label": "Sensitivity high",
+					"label": "High",
 					"value": 3
 				}
 			]
 		},
 		"8": {
 			"label": "Temperature Offset",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0371/zwa021.json:152]: Does this have an allowable range? Something like -50...50?

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/config/config/devices/0x0371/zwa009.json (hunk 11 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x0371/zwa009.json:45]: On the other hand, zwave-js should probably just give applications a way to display which CCs which Association Group sends. The device is new enough so it should support that.

Comment (AlCalzone) [reply packages/config/config/devices/0x0371/zwa009.json:45]: But we should at least change `MultiLevel` to `Multilevel`

File: packages/config/config/devices/0x0086/templates/aeotec_template.json (hunk 2 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0086/templates/aeotec_template.json]: Unintentional. I missed removing them.

File: packages/config/config/devices/0x0371/zw187.json (hunk 4 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0371/zw187.json]: Going with transmission. According to the manual it applies to Binary reports and notificaiton reports too so "command" seems weird.

File: packages/config/config/devices/0x0371/zw175.json (hunk 5 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0371/zw175.json]: Both. Duration or specifies which combination of buttons/reports.

File: packages/config/config/devices/0x0371/zw175.json (hunk 7 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0371/zw175.json:184]: Correct, this one the manual says the value per bit. I removed the "last setting" ones as they don't make much sense for bitmasks.

"Enable Hour:
Valid value are 0-23 and 127. Value 127 means use last valid setting.
Enable Minute:
Valid value are 0-59 and 255. Value 255 means use last valid setting"

File: packages/config/config/devices/0x0371/zwa005.json (hunk 8 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0371/zwa005.json:109]: They're different. One is magnet and one is motion and they're different values.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**